packages/react-native-storybook/ios/SherloModule.m

```diff
@@ -0,0 +1,305 @@
+- (NSDictionary *)constantsToExport {
+  return @{
+    @"syncDirectoryPath": sherloDirectoryPath,
+    @"initialMode": initialMode
+  };
+}
+
+RCT_EXPORT_METHOD(registered:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject) {
+    isStorybookRegistered = YES;
+
+    // Present the view controller replacing the root view controller, 
+    // the application root controller will never be shown
+    dispatch_async(dispatch_get_main_queue(), ^{
+      @try {
+        if (initialMode == @"testing") {
+          UIWindow *window = [UIApplication sharedApplication].windows.firstObject;
+          if (!window) {
+            return reject(@"E_NO_WINDOW", @"Failed to get the application window", nil);
+          }
+
+          UIViewController *rootViewController = window.rootViewController;
+          if (!rootViewController) {
+            return reject(@"E_NO_ROOTVC", @"No root view controller available", nil);
+          }
+
+          RCTRootView *rootView = [[RCTRootView alloc] initWithBridge:self.bridge moduleName:@"SherloStorybook" initialProperties:nil];
+          if (!rootView) {
+            return reject(@"E_CREATE_ROOTVIEW", @"Failed to create RCTRootView for Storybook", nil);
+          }
+
+          rootViewController.view = rootView;
+          [UIView transitionWithView:window duration:0.5 options:UIViewAnimationOptionTransitionCrossDissolve animations:nil completion:^(BOOL finished) {
+            if (!finished) {
+              return reject(@"E_NO_WINDOW", @"Failed to transition into storybook view", nil);
+            }
+          }];
+
+          resolve(nil);
+        }
+      } @catch (NSException *exception) {
+        [self handleException:exception rejecter:reject];
+      }
+    });
+}
+
+// Toggles the Storybook view. If the Storybook is currently open, it closes it. Otherwise, it opens it.
+// Exceptions during the toggle are caught and handled.
+RCT_EXPORT_METHOD(toggleStorybook:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject) {
+  if (currentStorybookViewController) {
+    [self closeStorybookInternal:resolve rejecter:reject];
+  } else {
+    [self openStorybookInternal:resolve rejecter:reject];
+  }
+}
+
+// Opens the Storybook view in a separate view controller on top of the root view controller,
+// allowing the user to return to the same app state after closing the Storybook.
+// Exceptions during the opening process are caught and handled.
+RCT_EXPORT_METHOD(openStorybook:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject) {
+  if(!isStorybookRegistered) {
+    return reject(@"NOT_REGISTERED", @"Storybook is not registered", nil);
+  }
+
+  [self openStorybookInternal:resolve rejecter:reject];
+}
+
+// Internal method to handle opening the Storybook view. This method is executed on the main queue.
+// If any errors occur, they are caught and the reject block is called with the appropriate error message.
```

Comment:
komentarz ze odpala sie to na dodatowym view controllerze z ktorego mozna nawigowac spowrotem